Smart person merge

apirogov opened the pull request.

mustafasoylu commented:
@apirogov 
For test purposes, I updated `pyproject.toml` line 5 as below and just changed a character in my family name. 
```toml
authors = ["Mustafa Doylu <m.soylu@fz-juelich.de>", "Anton Pirogov <a.pirogov@fz-juelich.de>"]
```

When I run `somesy sync`, the line remains as I changed it. Since `.somesy.toml` file has the right value, should not it `somesy` change it to the correct value?

apirogov commented:
> @apirogov For test purposes, I updated `pyproject.toml` line 5 as below and just changed a character in my family name.
> 
> ```toml
> authors = ["Mustafa Doylu <m.soylu@fz-juelich.de>", "Anton Pirogov <a.pirogov@fz-juelich.de>"]
> ```
> 
> When I run `somesy sync`, the line remains as I changed it. Since `.somesy.toml` file has the right value, should not it `somesy` change it to the correct value?

Nice catch! I only rigorously tested with the CITATION.cff writer... I'll check it.

apirogov commented:
@mustafasoylu You can try again, the issue was with model field aliases not being handled correctly (for CFF, we rely on aliases to parse fields with dashes)

mustafasoylu commented on src/somesy/core/writer.py:
```diff
@@ -150,7 +150,11 @@ def _merge_person_metadata(old: List[Person], new: List[Person]) -> List[Person]
                 # if there were changes -> update person
                 overlapping_fields = person.dict(include=set(person_update.keys()))
                 if person_update != overlapping_fields:
+                    print("updating ", modified_people[i], "->", person_update)
```
I think we should log these.

apirogov replied:
I forgot to remove them. Do you think we should actually debug log stuff in here anyway? I'm thinking it could be a bit too verbose.

mustafasoylu replied:
Since there might be a change to the outputs, I thought it would be nice to see the changes

apirogov replied:
Done

mustafasoylu reviewed (approved):
LGTM
